flask_api/app.py

```diff
@@ -0,0 +1,126 @@
+import logging
+from zipfile import BadZipFile
+from flask import Flask, request, jsonify
+import app_service
+from sqlalchemy.exc import IntegrityError
+
+app = Flask(__name__)
+
+# Configure the logger
+logging.basicConfig(level=logging.DEBUG)
+
+
+@app.route('/upload', methods=['POST'])
+def upload_file():
+    # Log request details
+    log_message = f"Request Details: Method={request.method}, URL={request.url}, IP={request.remote_addr}, User-Agent={request.user_agent}"
+    logging.info(log_message)
+
+    # check if file send
+    if 'file' not in request.files:
+        error_msg = 'No file found in request!'
+        return return_error_and_log(error_msg)
+
+    file = request.files['file']
+    email = request.form.get('email', None)
+
+    # Check if a file is selected
+    if file.filename == '':
+        error_msg = 'No file selected!'
+        return return_error_and_log(error_msg)
+
+    try:
+        upload = app_service.upload_pptx(file, user_email=email)
+    except (BadZipFile, ValueError) as e:
+        error_message = str(e)
+        return return_error_and_log(error_message)
+
+    logging.info('File uploaded successfully with UID: %s', upload['id'])
+    return jsonify(upload), 200
+
+
+@app.route('/status', methods=['GET'])
+def get_upload():
+    id = request.args.get('id', None)
+    email = request.args.get('email', None)
```

Comment:
you should validate the email

Reply:
i validate it the format and the mail existing in ij the table